src/canister_tests/src/tests.rs

```diff
@@ -24,16 +27,20 @@ fn ii_upgrade_retains_anchors() {
     let env = StateMachine::new();
     let canister_id = framework::install_ii_canister(&env, framework::II_WASM_PREVIOUS.clone());
     let challenge = api::create_challenge(&env, canister_id);
-    let user_number = match api::register(
-        &env,
+    let env_argument = &env;
+    let device_data = framework::some_device_data();
+    let challenge_attempt = types::ChallengeAttempt {
+        chars: "a".to_string(),
+        key: challenge.challenge_key,
+    };
+    let user_number = match register(
```

Comment:
how about `.unwrap()` or `.expect()`? in this test case I really don't think we ever expect an error, right?

Reply:
👍 Still have to match the variant though.